File: sample/game.lua
-- select tilemap.png as the tilemap
local tilemap = tmap('tilemap')
local font = fnt('MoroboxAIRetro')

local yFloor = 7 * 16
local xPlayer = 0
local yPlayer = yFloor
local isOnFloor = true
local xSpeed = 0.75
local xDir = 1
local xVel = 0
local yVel = 0
local time = 0

function tick(deltaTime)
    clear(0)
    tmode(16)
    
    time = time + deltaTime

    if btn(P1, BRIGHT) then
        xVel = xSpeed
    elseif btn(BLEFT) then
        xVel = -xSpeed
    else
        xVel = 0
    end

    if not isOnFloor then
        yVel = yVel + deltaTime * 0.25
        yPlayer = yPlayer + yVel * deltaTime

        if yPlayer >= yFloor then
            isOnFloor = true
            yPlayer = yFloor
        end
    end

    if btn(BUP) and isOnFloor then
        isOnFloor = false
        yVel = yVel - 5
    end

    xPlayer = xPlayer + xVel * deltaTime
    camera(xPlayer, SHEIGHT / 2)

    -- Background
    sclear()
    stile(tilemap, 0, 0, 1, 1)

    for i = -4, 18, 1 do
        sdraw(floor((xPlayer / 16) + i) * 16, 0)
        sdraw(floor((xPlayer / 16) + i) * 16, 7 * 16)
    end

    -- Player
    sclear()
    sflip(xDir < 0, false)
    sorigin(8, 16)

    if xVel ~= 0 then
        xDir = sign(xVel)
        stile(tilemap, 0 + floor((time % 16) / 8), 3, 1, 1)
    else
        stile(tilemap, 0, 3, 1, 1)
    end

    sdraw(xPlayer, yPlayer)

    -- Text
    tmode(8)
    sclear()
    stile(tilemap, 0, 8, 3, 3)
    sbox(8, 24, 96, 32)
    
    fclear()
    fcolor(0xFFFFFF)
    falign(0.5, 0.5)
    fdraw(font, "LUA SAMPLE", 56, 40)
end

File: sample/game.js
// select tilemap.png as the tilemap
tmap('tilemap');
// assign the tile (0, 3) to sprite 0
stile(0, 0, 3, 1, 1);
// center sprite 0 on screen
sorigin(0, 8, 16);

let yFloor = 7 * 16;
let xPlayer = 0;
let yPlayer = yFloor;
let isOnFloor = true;
let xSpeed = 1.5;
let xDir = 1;
let xVel = 0;
let yVel = 0;
let time = 0;

function tick(deltaTime) {
    time += deltaTime;

    if (btn(P1, BRIGHT)) {
        xVel = xSpeed;
    } else if (btn(BLEFT)) {
        xVel = -xSpeed;
    } else {
        xVel = 0;
    }

    if (!isOnFloor) {
        yVel += deltaTime * 0.25;
        yPlayer += yVel * deltaTime;

        if (yPlayer >= yFloor) {
            isOnFloor = true;
            yPlayer = yFloor;
        }
    }

    if (btn(BUP) && isOnFloor) {
        isOnFloor = false;
        yVel -= 5;
    }

    xPlayer = xPlayer + xVel * deltaTime;
    if (xVel !== 0) {
        xDir = sign(xVel);
        stile(0, 0 + floor((time % 16) / 8), 3, 1, 1);
    } else {
        stile(0, 0, 3, 1, 1);
    }

    sflip(0, xDir < 0, false);

    spos(0, SWIDTH / 2, yPlayer);

    const dX = xPlayer - (SWIDTH / 2);

    mclear();
    mscroll(dX, 0);

    mmode(16);
    for (let i = -2; i < 9; ++i) {
        mtile(floor(dX / 16) + i, 0, 0, 0);
        mtile(floor(dX / 16) + i, 7, 0, 0);
    }
}
